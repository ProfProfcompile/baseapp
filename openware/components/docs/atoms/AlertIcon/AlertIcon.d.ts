import * as React from 'react';
interface AlertIconProps {
    /**
     * Type of icon
     */
    type: 'success' | 'error';
}
declare const SuccessIcon: () => JSX.Element;
declare const ErrorIcon: () => JSX.Element;
declare const AlertIcon: React.FunctionComponent<AlertIconProps>;
export { AlertIcon, AlertIconProps, ErrorIcon, SuccessIcon, };
