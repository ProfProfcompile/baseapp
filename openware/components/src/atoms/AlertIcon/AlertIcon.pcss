.cr-alert-icon {
  height: var(--alert-icon-size);
  position: relative;
  width: var(--alert-icon-size);

  &--success {
    &__left,
    &__center,
    &__right {
      background: var(--base-2);
      border-radius: 4px;
      position: absolute;
      width: 3px;
    }

    &__left,
    &__right {
      height: 15px;
      top: 9px;
    }

    &__left {
      left: 4px;
      transform: rotate(-45deg);
    }

    &__center {
      border-radius: 0;
      left: 9px;
      height: 20px;
      top: 2px;
    }

    &__right {
      right: 5px;
      transform: rotate(45deg);
    }
  }

  &--error {
    &__left,
    &__right {
      background: var(--history-action-sell);
      border-radius: 4px;
      height: 27px;
      left: 10px;
      position: absolute;
      top: -3px;
      width: 3px;
    }

    &__left {
      transform: rotate(45deg);
    }

    &__right {
      transform: rotate(-45deg);
    }
  }
}
