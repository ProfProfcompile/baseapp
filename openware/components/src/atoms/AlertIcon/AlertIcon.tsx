import * as React from 'react';

interface AlertIconProps {
    /**
     * Type of icon
     */
    type: 'success' | 'error';
}

const SuccessIcon = () => (
    <div className="cr-alert-icon cr-alert-icon--success">
        <div className="cr-alert-icon--success__left" />
        <div className="cr-alert-icon--success__center" />
        <div className="cr-alert-icon--success__right" />
    </div>
);

const ErrorIcon = () => (
    <div className="cr-alert-icon cr-alert-icon--error">
        <div className="cr-alert-icon--error__left" />
        <div className="cr-alert-icon--error__right" />
    </div>
);

const AlertIcon: React.FunctionComponent<AlertIconProps> = (props: AlertIconProps) => {
    const { type } = props;

    return type === 'success'
        ? <SuccessIcon />
        : <ErrorIcon />;
};

export {
    AlertIcon,
    AlertIconProps,
    ErrorIcon,
    SuccessIcon,
};
