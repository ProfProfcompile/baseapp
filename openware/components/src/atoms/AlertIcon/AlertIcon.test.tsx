import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { AlertIcon, ErrorIcon, SuccessIcon } from './AlertIcon';

describe('AlertIcon', () => {
   it('should render success icon', () => {
        const wrapper: ShallowWrapper = shallow(<AlertIcon type="success" />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(SuccessIcon)).toHaveLength(1);
   });

   it('should render error icon', () => {
       const wrapper: ShallowWrapper = shallow(<AlertIcon type="error" />);
       expect(wrapper).toMatchSnapshot();
       expect(wrapper.find(ErrorIcon)).toHaveLength(1);
   });
});
