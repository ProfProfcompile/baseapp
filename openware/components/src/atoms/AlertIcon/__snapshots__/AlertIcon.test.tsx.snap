// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertIcon should render error icon 1`] = `<ErrorIcon />`;

exports[`AlertIcon should render success icon 1`] = `<SuccessIcon />`;
